Mathlib/Data/Set/Function.lean

```diff
@@ -1360,6 +1392,14 @@ theorem SurjOn.mapsTo_invFunOn [Nonempty α] (h : SurjOn f s t) : MapsTo (invFun
   fun _y hy => mem_preimage.2 <| invFunOn_mem <| h hy
 #align set.surj_on.maps_to_inv_fun_on Set.SurjOn.mapsTo_invFunOn
 
+theorem SurjOn.image_invFunOn_image_subset [Nonempty α] {r : Set β} (hf : SurjOn f s t)
+    (hrt : r ⊆ t) : f '' ((f.invFunOn s) '' r) = r :=
+  hf.rightInvOn_invFunOn.image_image' hrt
+
+theorem SurjOn.image_invFunOn_image [Nonempty α] (hf : SurjOn f s t) :
+    f '' ((f.invFunOn s) '' t) = t :=
+  hf.rightInvOn_invFunOn.image_image
+
```

Comment:
I was going to say that those lemmas should be replaced by `RightInvOn` statements as this is what the API expects you to use and I was going to suggest you the proofs. But I see you already found the proofs so you might as well drop the lemmas. We don't want one lemma of the form `f '' g '' s = s` for every possible pair of functions `f` and `g` that are inverse to each other on some set `s`. These lemmas are easily accessed on the fly using the API.

Reply:
Easily accessed only once you know they're there. I think this special case is prominent enough for these lemmas to exist so they're easily findable.

Reply:
Will let someone else weigh in

Reply:
Of course the lemmas are already there at the right generality, for me it is about the analogy between `InjOn` and `SurjOn`. It feels counterintuitive that for one, certain lemmas are there, but for others the lemmas that look like they should be there actually aren't. I think that it's best not to require that people figure out the reason for this. 

I'm also not sure how far through the API `exact?` can see in this case, but possibly this helps with that as well.